tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -216,6 +219,7 @@ def test_info(self):
     self.assertEqual(self.builder.name, info.name)
 
   def _get_dl_extract_result(self, url):
+    self.URLS.append(url)
```

Comment:
This might not be a string but a `tfds.download.Resource(`. Could you tried to run on `image/sun_test.py` to check this is working properly ?

Reply:
Yeah, just noticed that url may not be a string. 
```
{'images': <tensorflow_datasets.core.download.resource.Resource object at 0x7fb04a11c790>, 'partitions': 'https://vision.princeton.edu/projects/2010/SUN/download/Partitions.zip'}
```
But when I run this file (without including any of my changes)
```
python tensorflow_datasets/image/sun_test.py
```
I am getting segmentation fault. So, I will trying to resolve it first